docroot/modules/custom/sitenow_p2lb/sitenow_p2lb.module

```diff
@@ -1499,3 +1463,57 @@ function sitenow_p2lb_block_definition($type, array $fields) {
   $block_definition['type'] = $type;
   return array_merge($block_definition, $fields);
 }
+
+/**
+ * Get the protected revision id generated by p2lb.
+ *
+ * @param string $nid
+ *   The node id to get the p2lb protected revision id from.
+ *
+ * @return string|null
+ *   The protected revision id or null if none found.
+ *
+ * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+ * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+ */
+function sitenow_p2lb_get_protected_revision_id(string $nid) {
+  // Guard against the node id being null.
+  if (!$nid) {
+    return NULL;
+  }
+
+  // Load the node storage.
+  $node_storage = \Drupal::entityTypeManager()->getStorage('node');
+
+  // Get the node's most recent revision id.
+  $most_recent_revision_id = $node_storage?->getLatestRevisionId($nid);
+
+  // Load the most recent revision.
+  $most_recent_revision = $node_storage?->loadRevision($most_recent_revision_id);
+
+  // Get protected revision id from that revision's values.
+  // Return the protected revision id.
+  return $most_recent_revision?->field_v3_conversion_revision_id?->value;
+}
+
+/**
+ * Load a protected revision.
+ *
+ * @param string $revision_id
+ *   The revision id of the protected revision of a node.
+ *
+ * @return \Drupal\Core\Entity\RevisionableInterface|null
+ *   The Entity array for the protected revision or null if not found.
+ *
+ * @throws \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException
+ * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+ */
+function sitenow_p2lb_get_protected_revision(string $revision_id): ?RevisionableInterface {
```

Comment:
From the function name/comment, I was expecting this to take an `$nid` (or `$node`) and return the loaded protected revision (or `NULL` if there wasn't one), and not a specified revision.